packages/flutter_tools/lib/src/isolated/devfs_web.dart

```diff
@@ -878,19 +914,33 @@ class WebDevFS implements DevFS {
           'version.json', FlutterProject.current().getVersionInfo());
       webAssetServer.writeFile(
         'main.dart.js',
-        generateBootstrapScript(
-          requireUrl: 'require.js',
-          mapperUrl: 'stack_trace_mapper.js',
-          generateLoadingIndicator: enableDwds,
-        ),
+        ddcModuleSystem
+            ? generateDDCBootstrapScript(
+                entrypoint: entrypoint,
```

Comment:
If we don't have any configuration here to move `ddc_module_loader.js` into a subdirectory, I think it's pretty much guaranteed that the root directory for the scripts is handled by this entrypoint (and the result of `generateDDCMainModule`). Definitely no need to figure it out from JS!

Reply:
True, I agree. I think the directory handling logic 99% was for google3.